Invalid parsing of WASM linker flags

### Zig Version

0.11.0-dev.2546+cb54e9a3c

### Steps to Reproduce and Observed Behavior

Run:
```sh
zig cc main.c -Wl,--initial-memory=1048576
```

Output:
```
error: unsupported linker arg: --initial-memory
```

Run:
```sh
zig cc main.c -Wl,--shared-memory2
```

Output:
```
main.c:1:1: error: unable to build C object: FileNotFound
```

### Expected Behavior

The *first* linker flag, `--initial-memory`, is not recognized because an `=` is expected:
https://github.com/ziglang/zig/blob/602029bb2fb78048e46136784e717b57b8de8f2c/src/main.zig#L1962-L1973

But a previous step splits linker flags by `=`:
https://github.com/ziglang/zig/blob/602029bb2fb78048e46136784e717b57b8de8f2c/src/main.zig#L1675-L1690

Also suffering from this are: `--max-memory` and `--global-base`; `--export` works, because the alternative syntax is also accepted _here_ (there's another copy of this code where the alternative syntax is not explicitly accepted).

The *second* linker flag, `--shared-memory2`, is accepted and should be rejected. It's being compared with `mem.startsWith` and not `mem.eql`, not sure why.

--- a/src/main.zig
+++ b/src/main.zig
@@ -1425,7 +1425,7 @@ fn buildOutputType(
                         linker_initial_memory = parseIntSuffix(arg, "--initial-memory=".len);
                     } else if (mem.startsWith(u8, arg, "--max-memory=")) {
                         linker_max_memory = parseIntSuffix(arg, "--max-memory=".len);
-                    } else if (mem.startsWith(u8, arg, "--shared-memory")) {
+                    } else if (mem.eql(u8, arg, "--shared-memory")) {
                         linker_shared_memory = true;
                     } else if (mem.startsWith(u8, arg, "--global-base=")) {
                         linker_global_base = parseIntSuffix(arg, "--global-base=".len);
@@ -1959,16 +1959,23 @@ fn buildOutputType(
                     linker_import_table = true;
                 } else if (mem.eql(u8, arg, "--export-table")) {
                     linker_export_table = true;
-                } else if (mem.startsWith(u8, arg, "--initial-memory=")) {
-                    linker_initial_memory = parseIntSuffix(arg, "--initial-memory=".len);
-                } else if (mem.startsWith(u8, arg, "--max-memory=")) {
-                    linker_max_memory = parseIntSuffix(arg, "--max-memory=".len);
-                } else if (mem.startsWith(u8, arg, "--shared-memory")) {
+                } else if (mem.eql(u8, arg, "--initial-memory")) {
+                    const next_arg = linker_args_it.nextOrFatal();
+                    linker_initial_memory = std.fmt.parseUnsigned(u32, eatIntPrefix(next_arg, 16), 16) catch |err| {
+                        fatal("unable to parse initial memory size '{s}': {s}", .{ next_arg, @errorName(err) });
+                    };
+                } else if (mem.eql(u8, arg, "--max-memory")) {
+                    const next_arg = linker_args_it.nextOrFatal();
+                    linker_max_memory = std.fmt.parseUnsigned(u32, eatIntPrefix(next_arg, 16), 16) catch |err| {
+                        fatal("unable to parse max memory size '{s}': {s}", .{ next_arg, @errorName(err) });
+                    };
+                } else if (mem.eql(u8, arg, "--shared-memory")) {
                     linker_shared_memory = true;
-                } else if (mem.startsWith(u8, arg, "--global-base=")) {
-                    linker_global_base = parseIntSuffix(arg, "--global-base=".len);
-                } else if (mem.startsWith(u8, arg, "--export=")) {
-                    try linker_export_symbol_names.append(arg["--export=".len..]);
+                } else if (mem.eql(u8, arg, "--global-base")) {
+                    const next_arg = linker_args_it.nextOrFatal();
+                    linker_global_base = std.fmt.parseUnsigned(u32, eatIntPrefix(next_arg, 16), 16) catch |err| {
+                        fatal("unable to parse global base '{s}': {s}", .{ next_arg, @errorName(err) });
+                    };
                 } else if (mem.eql(u8, arg, "--export")) {
                     try linker_export_symbol_names.append(linker_args_it.nextOrFatal());
                 } else if (mem.eql(u8, arg, "--compress-debug-sections")) {

